speech/detector.go

```diff
@@ -322,6 +237,133 @@ func (sd *Detector) Detect(pcm []float32) ([]Segment, error) {
 	return segments, nil
 }
 
+type RealtimeSegment struct {
+	Start   int
+	End     int
+	Silence bool
+}
+
+// DetectRealtime returns the `cleaned` pcm (silence is fully zeroed out to improve transcriber accuracy),
+// and the voice and silence segments are detected and recorded in `segments`.
+func (sd *Detector) DetectRealtime(pcm []float32) (cleaned []float32, segments []RealtimeSegment, retErr error) {
```

Comment:
I feel this method is doing a bit too much and some of the logic should really live on the other side. In my mind this library should return enough metadata for the caller to do whatever is needed with the actual raw audio.

Ideally I'd like to extend the existing `Detect` method to return more of what we need to cover this use case.

Reply:
I agree, I'd like to simplify before releasing. But for now I'm keeping it because I focused on load testing and tuning the rest of the system. Once we have that working well I'll revisit this and clean it up before release.